k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java

```diff
@@ -1943,69 +1938,88 @@ public void loadLocalMessageForDisplay(LocalMessage message, Action action) {
         }
     }
 
-    class Listener extends MessagingListener {
+    private MessageLoaderCallbacks messageLoaderCallbacks = new MessageLoaderCallbacks() {
         @Override
-        public void loadMessageForViewStarted(Account account, String folder, String uid) {
-            if (mMessageReference == null || !mMessageReference.getUid().equals(uid)) {
-                return;
-            }
+        public void onMessageDataLoadFinished(LocalMessage message) {
+            // nothing to do here, we don't care about message headers
+        }
 
-            mHandler.sendEmptyMessage(MSG_PROGRESS_ON);
+        @Override
+        public void onMessageDataLoadFailed() {
+            mHandler.sendEmptyMessage(MSG_PROGRESS_OFF);
+            Toast.makeText(MessageCompose.this, R.string.status_invalid_id_error, Toast.LENGTH_LONG).show();
         }
 
         @Override
-        public void loadMessageForViewFinished(Account account, String folder, String uid, LocalMessage message) {
-            if (mMessageReference == null || !mMessageReference.getUid().equals(uid)) {
-                return;
-            }
+        public void onMessageViewInfoLoadFinished(LocalMessage localMessage, MessageViewInfo messageViewInfo) {
+            mHandler.sendEmptyMessage(MSG_PROGRESS_OFF);
+            loadLocalMessageForDisplay(localMessage, mAction);
+        }
 
+        @Override
+        public void onMessageViewInfoLoadFailed(LocalMessage localMessage) {
             mHandler.sendEmptyMessage(MSG_PROGRESS_OFF);
+            Toast.makeText(MessageCompose.this, R.string.status_invalid_id_error, Toast.LENGTH_LONG).show();
         }
 
         @Override
-        public void loadMessageForViewBodyAvailable(Account account, String folder, String uid,
-                final LocalMessage message) {
-            if (mMessageReference == null || !mMessageReference.getUid().equals(uid)) {
-                return;
+        public void setLoadingProgress(int current, int max) {
+            // nvm - we don't have a progress bar
```

Comment:
Sort of, but mostly it's indifference for the scope of this PR. I only deal with MessageCompose as much as necessary for it to work with MessageLoaderHelper here, introducing a progress bar for that activity may or may not be in a later PR.